### PR TITLE
onnxruntime 1.7.2 build and documentation

### PR DESCRIPTION
**Pull request by chayim**
Corrected how we build onnx. We can now build and release on x64, and x64 gpu. This will build on osx x64 as well, and version 1.8 of onnx now builds (using these changes) on jetson. This PR is only for 1.7.2. A new developer should be able to use the accompanying instructions, in order to work on the onnx build should they so desire.

This would be the template for building back-ends.  [ci was built in full](https://app.circleci.com/pipelines/github/RedisAI/RedisAI/3791/workflows/b6a8117e-985f-4b30-a627-1907a56a8a96) and the remaining run is for changing circleci back to master's norm.

### CODE REVIEW

File: opt/build/backends.rules [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		-t ${DEFAULT_DOCKER_TAG} \
+        -S ../dockerparts \
+		-e REDIS \
+		-D "${DOCKER_OPTS}" \
+		${DOCKER_ARGS}
+	docker create --cidfile ${CIDFILE} ${DEFAULT_DOCKER_TAG}
+	docker cp `cat ${CIDFILE}`:/build/${PRODUCT}/${BACKEND_NAME} .
+
+publish:
+	@aws s3 cp ${BACKEND_NAME} s3://$(S3_URL)/ --acl public-read
```
Comment (DvirDukhan) [inline opt/build/backends.rules:48]: do we want to expose this?

File: opt/build/onnxruntime/Makefile [hunk 2]
```diff
@@ -1,134 +1,16 @@
+REDIS_ONNX_VERSION?=rel-1.7.2_RCE
```
Comment (DvirDukhan) [inline opt/build/onnxruntime/Makefile:1]: this is the branch name?

File: opt/build/onnxruntime/dockerfile.tmpl [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#----------------------------------------------------------------------------------------------
+{% if REDIS_ARCH == 'x64' %}
+{% if REDIS_GPU is defined %}
+FROM nvidia/cuda:11.0-cudnn8-devel-ubuntu18.04
```
Comment (DvirDukhan) [inline opt/build/onnxruntime/dockerfile.tmpl:4]: maybe allow user to define their cuda version?
we use 11.0
might want to compile it on newer versions

**Comment by alonre24** [comment]
moved to #847

File: opt/build/backends.rules (hunk 1 above)
Comment (chayim) [reply opt/build/backends.rules:48]: There's nothing being exposed here...

**Pull request closed without merging**